Documentation/scalability/scalability.rst

```diff
@@ -0,0 +1,397 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here we have created 125 L4 network policies and 125 L7 policies. Each policy
+selected all pods on this namespace and was allowed to send traffic to another
+pod on this namespace. Each 250 policies differ on the chosen port. In the end
+we will have 250 different policies selecting 10000 pods.
+
+::
+
+    apiVersion: "cilium.io/v2"
+    kind: CiliumNetworkPolicy
+    metadata:
+      name: "l4-rule-#"
+      namespace: "namespace-1"
+    spec:
+      endpointSelector:
+        matchLabels:
+          my-label: testing
+      fromEndpoints:
+        matchLabels:
+          my-label: testing
+      egress:
+        - toPorts:
+          - ports:
+            - port: "[0-125]+80" // from 80 to 12580
+              protocol: TCP
+    ---
+    apiVersion: "cilium.io/v2"
+    kind: CiliumNetworkPolicy
+    metadata:
+      name: "l7-rule-#"
+      namespace: "namespace-1"
+    spec:
+      endpointSelector:
+        matchLabels:
+          my-label: testing
+      fromEndpoints:
+        matchLabels:
+          my-label: testing
+      ingress:
+      - toPorts:
+        - ports:
+          - port: '[126-250]+80' // from 12680 to 25080
+            protocol: TCP
+          rules:
+            http:
+            - method: GET
+              path: "/path1$"
+            - method: PUT
+              path: "/path2$"
+              headers:
+              - 'X-My-Header: true'
+
+.. figure:: images/image_8_01.png
+
+In this case we saw one of the Cilium agents jumping to 100% CPU usage for 15
+seconds while the average peak was 40% during a period of 90 seconds.
+
+.. figure:: images/image_8_02.png
+
+As expected, **increasing the number of policies does not have a significant
+impact in the memory usage of Cilium since the BPF policy maps have a constant
+size** once a pod is initialized.
+
+.. figure:: images/image_8_03.png
+.. figure:: images/image_8_04.png
+
+
+The first mark represents the point in time when we ran ``kubectl create`` to
+create the ``CiliumNetworkPolicies``. Since we created 250 policies sequentially
+we cannot correctly derive the time between the creation of the 250th CNP and
+the time the last endpoint had its policy enforced. In our examples we have
+used 250 different CNP objects, to correctly measure the convergence time we
+could have used a single CNP with multiple policy rules defined under the
+``specs`` field (not ``spec``) for a CNP.
+
+However we can see the time it took the last Cilium agent to increment its
+Policy Revision, which is incremented individually on each Cilium agent every
+time time a CiliumNetworkPolicy (CNP) is received, between second ``15:45:44``
+and ``15:45:46`` and see when was the last time an Endpoint was regenerated by
+checking the 99th percentile of the "Endpoint regeneration time", we can see
+that it took less than 5s. We can also verify **the maximum time was less than
+``600ms`` for an endpoint to have the policy enforced.**
```

Comment:
Note that the ` ``600ms`` ` is not rendered correctly on the preview, most likely because it is nested inside the markup for bold text.